pkg/graph/types.go

```diff
@@ -116,19 +123,40 @@ func (v *Vertex) NewWithSameRef() *Vertex {
 	}
 }
 
-func (v *Vertex) AddEdge(to *Vertex, edgeRef *duckv1.Destination, transform TransformFunction) {
-	edge := &Edge{from: v, to: to, transform: transform, self: edgeRef}
+func (v *Vertex) AddEdge(to *Vertex, edgeRef *duckv1.Destination, transform Transform, isDLS bool) {
```

Comment:
Eventually, my recommendation would be to refactor this method and pass struct instead of raw fields, it is easier to evolve and avoid breaking changes.

Also it helps with being easier to read on the caller side

```
v.AddEdge(to, dest, NoTransform{}, true)
```

Reply:
Do you mean something like:
```go
v.AddEdge(edge)
```
@pierDipi ?

Reply:
yes